task/test-github

```diff
@@ -111,19 +111,19 @@ def mockServer():
     httpd.data = data
 
     child = os.fork()
-    if child != 0:
-        httpd.server_close()
-        return child
-
-    # prctl(PR_SET_PDEATHSIG, SIGTERM)
-    try:
-        libc = ctypes.CDLL('libc.so.6')
-        libc.prctl(1, 15)
-    except OSError:
-        pass
-
-    httpd.serve_forever()
-    os._exit(1)
+    if child == 0:
```

Comment:
This code is horrifying, considering why codeql thinks that we might return from this function in the child (and I can consider another case: an exception).

Can you change this to use a thread or `multiprocessing` module or something?

Reply:
Resolved in #4050